plugins/vendor/zoom/ZoomDropFolderPlugin/batch/processors/zoomRecordingProcessor.php

```diff
@@ -323,11 +339,88 @@ protected function getAdditionalUsers($recordingId, $userToExclude)
 		return $this->getValidatedUsers($additionalZoomUsers, $this->dropFolder->partnerId, $this->dropFolder->zoomVendorIntegration->createUserIfNotExist,
 		                                $userToExclude);
 	}
-	
+
+	/**
+	 * @param string $recordingId
+	 * @param string $userToExclude
+	 * @return array|null
+	 */
+	protected function getValidatedHosts($recordingId, $userToExclude)
+	{
+		$userToExclude = strtolower($userToExclude);
+		$zoomCoHosts = $this->getCoHostsEmails($recordingId);
+		$zoomAlternativeHosts = $this->getAlternativeHostsEmails($recordingId);
+		$zoomHosts = $zoomCoHosts ? array_merge($zoomCoHosts, $zoomAlternativeHosts) : $zoomAlternativeHosts;
+		$zoomHosts = $zoomHosts ? $zoomHosts : null;
+		return $this->getValidatedUsers($zoomHosts, $this->dropFolder->partnerId, $this->dropFolder->zoomVendorIntegration->createUserIfNotExist,
+										$userToExclude);
+	}
+
+	protected function getAlternativeHostsEmails($recordingId)
+	{
+		$data = $this->getAlternativeHostsData($recordingId);
+		if(isset($data[self::SETTINGS]) && isset($data[self::SETTINGS][self::ALTERNATIVE_HOSTS]))

```

Comment:
if an account doesn't have the relevant scope it will fail here, right?

Reply:
this happens for cohosts in plugins/vendor/zoom/ZoomDropFolderPlugin/batch/processors/zoomRecordingProcessor.php:371
zoom api fails but we don't (we treat it as empty)